test/unit/test_tpm2_eventlog.c

```diff
@@ -42,15 +45,50 @@ static void test_foreach_digest2(void **state) {
     TCG_DIGEST2* digest = (TCG_DIGEST2*)buf;
 
     will_return(foreach_digest2_test_callback, true);
-    assert_true(foreach_digest2(digest, 1, TCG_DIGEST2_SHA1_SIZE, foreach_digest2_test_callback, NULL));
+
+    tpm2_eventlog_ctx_t ctx = { .digest2_cb = foreach_digest2_test_callback };
+    assert_true(foreach_digest2(&ctx, 0, digest, 1, TCG_DIGEST2_SHA1_SIZE));
 }
 static void test_foreach_digest2_cbnull(void **state){
 
     (void)state;
-    uint8_t buf [TCG_DIGEST2_SHA1_SIZE] = { 0, };
+    uint8_t buf [TCG_DIGEST2_SHA1_SIZE] = { };
     TCG_DIGEST2* digest = (TCG_DIGEST2*)buf;
 
-    assert_true(foreach_digest2(digest, 1, TCG_DIGEST2_SHA1_SIZE, NULL, NULL));
+    tpm2_eventlog_ctx_t ctx = {};
+    assert_true(foreach_digest2(&ctx, 0, digest, 1, TCG_DIGEST2_SHA1_SIZE));
+}
+static void test_sha1(void **state){
+
+    (void)state;
+    uint8_t buf [TCG_DIGEST2_SHA1_SIZE] = { 0, };
+    uint8_t sha1sum[] = { 0x31,0x19,0x5d,0x69,0x35,0x16,0x3c,0x79,0xa9,0x67,0x22,0xba,0x7d,0x4b,0x11,0x35,0x24,0x89,0xf4,0x8b };
+    const int pcrIndex = 3;
```

Comment:
No CamelCasing pcr_index. (I know the code base is not 100% consistent, but lets not add to the issue)

Reply:
I was following the structures, which are CamelCased.  Fixed!